core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -130,71 +121,71 @@ angular.module('oppia').directive('adminMiscTab', [
         };
 
         ctrl.sendDummyMailToAdmin = function() {
-          $http.post(SEND_DUMMY_MAIL_HANDLER_URL)
+          AdminMiscTabBackendApiService.sendDummyMail()
             .then(function(response) {
               ctrl.setStatusMessage('Success! Mail sent to admin.');
             }, function(errorResponse) {
+              console.log(errorResponse)
               ctrl.setStatusMessage(
-                'Server error: ' + errorResponse.data.error);
+                'Server error: ' + errorResponse.error.error);
             });
         };
 
         ctrl.flushMemoryCache = function() {
-          $http.post(MEMORY_CACHE_HANDLER_URL)
+          AdminMiscTabBackendApiService.flushCache()
             .then(function(response) {
               ctrl.setStatusMessage('Success! Memory Cache Flushed.');
             }, function(errorResponse) {
               ctrl.setStatusMessage(
-                'Server error: ' + errorResponse.data.error);
+                'Server error: ' + errorResponse.error.error);
             });
         };
 
         ctrl.getMemoryCacheProfile = function() {
-          $http.get(MEMORY_CACHE_HANDLER_URL)
+          AdminMiscTabBackendApiService.getMemoryCacheProfile()
             .then(function(response) {
```

Comment:
Service should return a domain object not the response dict*

Reply:
done!